File: Leetcode/sbgo/Leetcode1614-括号的最大嵌套深度.go
package sbgo

func maxDepth(s string) int {
	ans := 0
	for i, cnt := 0, 0; i < len(s); i++ {
		if s[i] == '(' {
			cnt++
			if ans < cnt {
				ans = cnt
			}
		} else if s[i] == ')' {
			cnt--
		}
	}
	return ans
}

File: Leetcode/Rust/Leetcode1614-括号的最大嵌套深度.rs
impl Solution {
    pub fn max_depth(s: String) -> i32 {
        let (mut ans,mut cnt) = (0,0);
        for ch in s.chars(){
            if ch == '('{
                cnt += 1;
                ans = ans.max(cnt);
            }else if ch ==')'{
                cnt -= 1;
            }
        }
        ans
    }
}